crates/modelardb_server/tests/integration_test.rs

```diff
@@ -41,120 +40,360 @@ use sysinfo::{Pid, PidExt, System, SystemExt};
+            self.client.do_action(Request::new(action)).await.unwrap();
+        })
+    }
+
+    /// Execute a query against the server through the `do_get()` method and return it.
+    fn execute_query(&mut self, query: String) -> Result<Vec<RecordBatch>, Box<dyn Error>> {
+        self.runtime.block_on(async {
+            // Execute query.
+            let ticket = Ticket {
+                ticket: query.into(),
+            };
+            let mut stream = self.client.do_get(ticket).await?.into_inner();
+
+            // Get schema of result set.
+            let flight_data = stream.message().await?.ok_or("No data_points received.")?;
+            let schema = Arc::new(Schema::try_from(&flight_data)?);
+
+            // Get data in result set.
+            let mut results = vec![];
+            while let Some(flight_data) = stream.message().await? {
+                let dictionaries_by_id = HashMap::new();
+                let record_batch = utils::flight_data_to_arrow_batch(
+                    &flight_data,
+                    schema.clone(),
+                    &dictionaries_by_id,
+                )?;
+                results.push(record_batch);
+            }
+            Ok(results)
+        })
+    }
+
+    /// Retrieve the table names currently in the server and return them.
+    fn retrieve_all_table_names(&mut self) -> Result<Vec<String>, Box<dyn Error>> {
+        let criteria = Criteria {
+            expression: Bytes::new(),
+        };
+        let request = Request::new(criteria);
+
+        self.runtime.block_on(async {
+            let mut stream = self.client.list_flights(request).await?.into_inner();
+            let flights = stream.message().await?.ok_or("No data_points received.")?;
+
+            let mut table_names = vec![];
+            if let Some(fd) = flights.flight_descriptor {
+                for table in fd.path {
+                    table_names.push(table);
+                }
+            }
+            Ok(table_names)
+        })
+    }
+
+    /// Retrieve the schema of a table in the server and return it.
+    fn retrieve_schema(&mut self, table_name: &str) -> Schema {
+        self.runtime.block_on(async {
+            let schema_result = self
+                .client
+                .get_schema(Request::new(FlightDescriptor::new_path(vec![
+                    table_name.to_owned()
+                ])))
+                .await
+                .unwrap()
+                .into_inner();
+
+            convert::try_schema_from_ipc_buffer(&schema_result.schema).unwrap()
+        })
+    }
+}
+
+impl Drop for TestContext {
+    /// Kill the server process when [`TestContext`] is dropped.
+    fn drop(&mut self) {
+        let mut system = System::new_all();
+
+        while let Some(_process) = system.process(Pid::from_u32(self.server.id())) {
+            system.refresh_all();
+
+            // Microsoft Windows often fail to kill the process and then succeed afterwards.
```

Comment:
no need to give up after n attempts?

Reply:
Added a counter that exits the loop after ten attempts. Ten was chosen arbitrarily.